pytorch/pytorch_lightning_ddp.py

```diff
@@ -123,21 +123,24 @@ def objective(trial: optuna.trial.Trial) -> float:
 
     model = LightningNet(dropout, output_dims)
     datamodule = FashionMNISTDataModule(data_dir=DIR, batch_size=BATCHSIZE)
+    callback = PyTorchLightningPruningCallback(trial, monitor="val_acc")
 
     trainer = pl.Trainer(
         logger=True,
         limit_val_batches=PERCENT_VALID_EXAMPLES,
         enable_checkpointing=False,
         max_epochs=EPOCHS,
-        gpus=-1 if torch.cuda.is_available() else None,
-        accelerator="ddp_cpu" if not torch.cuda.is_available() else None,
-        num_processes=os.cpu_count() if not torch.cuda.is_available() else None,
-        callbacks=[PyTorchLightningPruningCallback(trial, monitor="val_acc")],
+        accelerator="auto" if torch.cuda.is_available() else "cpu",
+        devices="auto" if not torch.cuda.is_available() else os.cpu_count(),
```

Comment:
Currently, devices will be the number of CPUs if CUDA is available. I guess it should be the number of GPUs. Or, can we always set `auto` here?

```suggestion
        devices="auto",
```

Reply:
You're right. `devices` has to be less than or equals to the number of available `Accelerators`. I think
```
devices="auto" if torch.cuda.is_available() else os.cpu_count(),
```
would be better since `Trainer` uses `devices=1` by default when `accelerator="cpu"`.
https://lightning.ai/docs/pytorch/stable/common/trainer.html#:~:text=%22tpu%22)-,TIP,-The%20%22auto%22

Reply:
Thank you for your confirmation. The latest code (https://github.com/optuna/optuna-examples/pull/179/commits/afb1e13d1bde37905acc26cf72888109e2e8060b) fixed the number of devices to 2, and I think it is reasonable since users usually have more than two devices to train models in parallel.

Reply:
Thank you for your comment. As we discussed offline, we decided to set `devices=2`. This is because `devices=auto` results in 
1. `devices=1` when `accelerator="cpu"`
2. too many `devices` for the dataset to split might be set when `accelerator` is not `"cpu"`.

Reply:
Sounds reasonable to me. Thanks!